src/io/flutter/inspector/InspectorGroupManagerService.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.inspector;
+
+import com.google.common.collect.Lists;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
+import io.flutter.run.FlutterAppManager;
+import io.flutter.run.daemon.FlutterApp;
+import io.flutter.utils.AsyncUtils;
+import org.dartlang.vm.service.VmService;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Service that provides the current inspector state independent of a specific
+ * application run.
+ * <p>
+ * This class provides a {@link Listener} that notifies consumers when state
+ * has changed.
+ * <ul>
+ * <li>The inspector is available.</li>
+ * <li>The inspector selection when it changes</li>
+ * <li>When the UI displayed in the app has changed</li>
+ * </ul>
+ */
+public class InspectorGroupManagerService implements Disposable {
```

Comment:
I'm not the right reviewer for this file.

Reply:
@kenzieschmoll  can you review this file? You might be slightly more familiar with this code.